charm/kafka/reactive/nagios.py

```diff
@@ -0,0 +1,115 @@
+import os
+import shutil
+
+from charmhelpers.core import hookenv
+
+from charms.reactive import when, when_not, set_state
+
+from charms.layer.kafka import KAFKA_SNAP
+
+
+@when('local-monitors.available')
+def local_monitors_available(nagios):
+    setup_nagios(nagios)
+
+
+@when('nrpe-external-master.available')
+def nrpe_external_master_available(nagios):
+    setup_nagios(nagios)
+
+
+def setup_nagios(nagios):
+    config = hookenv.config()
+    unit_name = hookenv.local_unit()
+    checks = [{
+        'name': 'underreplicated_partitions_count',
+        'object_name': 'kafka.server:type=ReplicaManager,\
+name=UnderReplicatedPartitions',
+        'description': 'Number of under replicated partitions',
+        'crit': '!= 0',
+    }, {
+        'name': 'active_controller_count',
+        'object_name': 'kafka.controller:type=KafkaController,\
+name=ActiveControllerCount',
+        'description': 'Number of active controllers in cluster',
+        'crit': '!= 1'
+    }, {
+        'name': 'offline_partitions_count',
+        'object_name': 'kafka.controller:type=KafkaController,\
+name=OfflinePartitionsCount',
+        'description': 'Number of offline partitions in cluster',
+        'crit': '!= 1'
```

Comment:
Would this be crit if > 0? != 0 is fine as well, can't imagine negative offline partitions would be something you'd want to see either...